From 05e6e8fd52a029789d7a1a400ce54d096d9925b7 Mon Sep 17 00:00:00 2001
From: Peter Hutterer <peter.hutterer@who-t.net>
Date: Tue, 15 Jun 2010 10:16:38 +1000
Subject: [PATCH] man: update SHMConfig documentation, it's only useful for debugging now.

Signed-off-by: Peter Hutterer <peter.hutterer@who-t.net>
Reviewed-by: Fernando Carrijo <fcarrijo@yahoo.com.br>
(cherry picked from commit 1f2b54cd57e88b706b6b2874547eabd1a8ab0963)

man: divide up man page with a few subheaders.

Signed-off-by: Peter Hutterer <peter.hutterer@who-t.net>
Reviewed-by: Fernando Carrijo <fcarrijo@yahoo.com.br>
(cherry picked from commit 78df6c5da2792a333a7aac899329517032f2f2f8)

man: move area description down to CONFIGURATION DETAILS section.

Signed-off-by: Peter Hutterer <peter.hutterer@who-t.net>
Reviewed-by: Fernando Carrijo <fcarrijo@yahoo.com.br>
(cherry picked from commit f42fa446c4c0a2de92c1b5320dc8dcd0c18907e5)

man: note where the Area options fit in.

Signed-off-by: Peter Hutterer <peter.hutterer@who-t.net>
Reviewed-by: Fernando Carrijo <fcarrijo@yahoo.com.br>
(cherry picked from commit 857d579de808a95e3c8541847ec8a8845ca57c73)

man: don't call it "synaptics" touchpad driver.

Under Linux the kernel handles everything, so these days synaptics is the
generic touchpad driver for anything that the kernel can handle. There
aren't many synaptics-specific bits in the driver anymore.

Signed-off-by: Peter Hutterer <peter.hutterer@who-t.net>
Reviewed-by: Fernando Carrijo <fcarrijo@yahoo.com.br>
(cherry picked from commit 8b3717055ab5c8d52bf5aae13e996ab4e86c2794)
---
 man/synaptics.man |  111 ++++++++++++++++++++++++++++------------------------
 1 files changed, 60 insertions(+), 51 deletions(-)

diff --git a/man/synaptics.man b/man/synaptics.man
index 59fbaac..44760a5 100644
--- a/man/synaptics.man
+++ b/man/synaptics.man
@@ -2,7 +2,7 @@
 .ds q \N'34'
 .TH SYNAPTICS __drivermansuffix__ __vendorversion__
 .SH NAME
-synaptics \- Synaptics touchpad input driver
+synaptics \- touchpad input driver
 .SH SYNOPSIS
 .nf
 .B "Section \*qInputDevice\*q"
@@ -15,10 +15,9 @@ synaptics \- Synaptics touchpad input driver
 .fi
 .SH DESCRIPTION
 .B synaptics
-is an __xservername__ input driver for the touchpads from Synaptics
-Incorporated. Even though these touchpads (by default, operating in a
-compatibility mode emulating a standard mouse) can be handled by the normal
-evdev or mouse drivers, this driver allows more advanced features of the
+is an __xservername__ input driver for touchpads.
+Even though touchpads can be handled by the normal evdev or mouse drivers,
+this driver allows more advanced features of the
 touchpad to become available. Some benefits would be:
 .IP \(bu 4
 Movement with adjustable, non-linear acceleration and speed.
@@ -64,7 +63,12 @@ that some functions are not available on all touchpad models, because
 they need support from the touchpad hardware/firmware. (Multifinger
 taps for example.)
 .PP
-.SH CONFIGURATION DETAILS
+The name "synaptics" is historical and the driver still provides the
+synaptics protocol parsing code. Under Linux however, the hardware-specifics
+are handled by the kernel and this driver will work for any touchpad that
+has a working kernel driver.
+
+.SH CONFIGURATION OPTIONS
 Please refer to __xconfigfile__(__filemansuffix__) for general configuration
 details and for options that can be used with all input drivers.  This
 section only covers configuration details specific to this driver.
@@ -90,10 +94,9 @@ psm	FreeBSD psm driver
 .TE
 .TP 7
 .BI "Option \*qSHMConfig\*q \*q" boolean \*q
-Switch on/off shared memory for run-time configuration. Note that this is
-considered a security risk since any user can access the configuration. This
-option is not needed with synaptics 1.0 or later. See section
-.B Device Properties.
+Switch on/off shared memory for run-time debugging. This option does not
+have an effect on run-time configuration anymore and is only useful for
+hardware event debugging.
 .TP 7
 .BI "Option \*qLeftEdge\*q \*q" integer \*q
 X coordinate for left edge. Property: "Synaptics Edges"
@@ -442,7 +445,38 @@ horizontal sensitivity. Setting VertResolution and HorizResolution
 equal values means no compensation. Default value is read from
 the touchpad or set to 1 if value could not be read.
 Property: "Synaptics Pad Resolution"
-.LP
+.
+.TP
+.BI "Option \*qAreaLeftEdge\*q \*q" integer \*q
+Ignore movements, scrolling and tapping which take place left of this edge.
+.
+The option is disabled by default and can be enabled by setting the
+AreaLeftEdge option to any integer value other than zero. Property: "Synaptics Area"
+.
+.TP
+.BI "Option \*qAreaRightEdge\*q \*q" integer \*q
+Ignore movements, scrolling and tapping which take place right of this edge.
+.
+The option is disabled by default and can be enabled by setting the
+AreaRightEdge option to any integer value other than zero. Property: "Synaptics Area"
+.
+.TP
+.BI "Option \*qAreaTopEdge\*q \*q" integer \*q
+Ignore movements, scrolling and tapping which take place above this edge.
+.
+The option is disabled by default and can be enabled by setting the
+AreaTopEdge option to any integer value other than zero. Property: "Synaptics Area"
+.
+.TP
+.BI "Option \*qAreaBottomEdge\*q \*q" integer \*q
+Ignore movements, scrolling and tapping which take place below this edge.
+.
+The option is disabled by default and can be enabled by setting the
+AreaBottomEdge option to any integer value other than zero. Property: "Synaptics Area"
+.
+
+.SH CONFIGURATION DETAILS
+.SS Area handling
 The LeftEdge, RightEdge, TopEdge and BottomEdge parameters are used to
 define the edge and corner areas of the touchpad.
 .
@@ -476,40 +510,17 @@ Physical left edge		Physical right edge
 Coordinates to the left of LeftEdge are part of the left edge (areas
 1, 4 and 7), coordinates to the left of LeftEdge and above TopEdge
 (area 1) are part of the upper left corner, etc.
-.
+.PP
 A good way to find appropriate edge parameters is to enable the
 SHMConfig option and run "synclient \-m 1" to see the x/y coordinates
 corresponding to different positions on the touchpad.
-.
-.TP
-.BI "Option \*qAreaLeftEdge\*q \*q" integer \*q
-Ignore movements, scrolling and tapping which take place left of this edge.
-.
-The option is disabled by default and can be enabled by setting the
-AreaLeftEdge option to any integer value other than zero. Property: "Synaptics Area"
-.
-.TP
-.BI "Option \*qAreaRightEdge\*q \*q" integer \*q
-Ignore movements, scrolling and tapping which take place right of this edge.
-.
-The option is disabled by default and can be enabled by setting the
-AreaRightEdge option to any integer value other than zero. Property: "Synaptics Area"
-.
-.TP
-.BI "Option \*qAreaTopEdge\*q \*q" integer \*q
-Ignore movements, scrolling and tapping which take place above this edge.
-.
-The option is disabled by default and can be enabled by setting the
-AreaTopEdge option to any integer value other than zero. Property: "Synaptics Area"
-.
-.TP
-.BI "Option \*qAreaBottomEdge\*q \*q" integer \*q
-Ignore movements, scrolling and tapping which take place below this edge.
-.
-The option is disabled by default and can be enabled by setting the
-AreaBottomEdge option to any integer value other than zero. Property: "Synaptics Area"
-.
-.LP
+.PP
+The perceived physical edges may be adjusted with the AreaLeftEdge,
+AreaRightEdge, AreaTopEdge, and AreaBottomEdge options. If these values are
+set to something other than the physical edges, input in the space between
+the area edge and the respective physical edge is ignored. Note that this
+reduces the available space on the touchpad.
+.SS Tapping
 A tap event happens when the finger is touched and released in a time
 interval shorter than MaxTapTime, and the touch and release
 coordinates are less than MaxTapMove units apart.
@@ -536,8 +547,7 @@ too small value means that visual feedback from the gui application
 you are interacting with is harder to see.
 .
 For this parameter to have any effect, "FastTaps" has to be disabled.
-.
-.LP
+.SS Acceleration
 The MinSpeed, MaxSpeed and AccelFactor parameters control the pointer
 motion speed.
 .
@@ -606,8 +616,7 @@ EdgeMotionMinZ and EdgeMotionMaxZ.
 .
 For a pressure value between PressureMotionMinZ and
 PressureMotionMaxZ, the factor is increased linearly.
-.
-.LP
+.SS Middle button emulation
 Since most synaptics touchpad models don't have a button that
 corresponds to the middle button on a mouse, the driver can emulate
 middle mouse button events.
@@ -616,7 +625,7 @@ If you press both the left and right mouse buttons at almost the same
 time (no more than EmulateMidButtonTime milliseconds apart) the driver
 generates a middle mouse button event.
 .
-.LP
+.SS Circular scrolling
 Circular scrolling acts like a scrolling wheel on the touchpad.
 .
 Scrolling is engaged when a drag starts in the given CircScrollTrigger
@@ -635,8 +644,8 @@ large circles for better control.
 When used together with vertical scrolling, hitting the upper or lower
 right corner will seamlessly switch over from vertical to circular
 scrolling.
-.
-.LP
+
+.SS Coasting
 Coasting is enabled by setting the CoastingSpeed parameter to a
 non-zero value.
 .
@@ -672,8 +681,8 @@ CornerCoasting takes precedence over the seamless switch from edge
 scrolling to circular scrolling.  That is, if CornerCoasting is
 active, scrolling will stop, and circular scrolling will not start,
 when the finger leaves the corner.
-.
-.LP
+
+.SS Trackstick mode
 Trackstick emulation mode is entered when pressing the finger hard on
 the touchpad.
 .
-- 
1.6.5.2

